cmd/state-tray/internal/open/open_lin.go

```diff
@@ -28,3 +28,38 @@ func Prompt(command string) error {
 
 	return nil
 }
+
+func preferredShellWithFallback(fallback string) (string, error) {
+	currentUser, err := user.Current()
+	if err != nil {
+		return "", locale.WrapError(err, "err_user_unknown", "Cannot get current user")
+	}
+
+	f, err := os.Open("/etc/passwd")
+	if err != nil {
+		return "", locale.WrapError(err, "err_open_passwd", "Cannot open passwd file")
+	}
+	defer f.Close()
+
+	prefix := currentUser.Name + ":"
+	var shell string
+
+	sc := bufio.NewScanner(f)
+	for sc.Scan() {
+		line := sc.Text()
+		if !strings.HasPrefix(line, prefix) {
+			continue
+		}
+		parts := strings.Split(line, ":")
+		shell = parts[len(parts)-1]
+	}
+	if err := sc.Err(); err != nil {
+		return "", locale.WrapError(err, "err_scan_passwd", "/etc/passwd file scan failed")
```

Comment:
Explain why we're looking at this.

Reply:
Also changed this one to not be as alarming.

Reply:
I did add a comment for anyone reading the code, however.